File: src/main/java/dev/paigewatson/layoutmaster/client/controllers/GoodsController.java
package dev.paigewatson.layoutmaster.client.controllers;

import dev.paigewatson.layoutmaster.models.goods.GoodsType;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.List;

@RestController
@CrossOrigin
@RequestMapping("/models")
public class GoodsController
{

    @GetMapping(value = "/goods")
    public List<GoodsType> getAllGoods()
    {
        return Arrays.asList(GoodsType.class.getEnumConstants());
    }
}

File: src/main/java/dev/paigewatson/layoutmaster/client/controllers/GoodsController.kt
package dev.paigewatson.layoutmaster.client.controllers

import dev.paigewatson.layoutmaster.models.goods.GoodsType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.*

@RestController
@RequestMapping("/models")
class GoodsController {
    @get:GetMapping(value = ["/goods"])
    val allGoods: List<GoodsType>
        get() = Arrays.asList(*GoodsType::class.java.enumConstants)
}
